src/modules/0.0.21/aws_appsync/index.ts

```diff
@@ -0,0 +1,180 @@
+    else return undefined;
+  }
+
+  getGraphqlApi = crudBuilderFormat<AppSync, 'getGraphqlApi', GraphqlApiAWS | undefined>(
+    'getGraphqlApi',
+    apiId => apiId,
+    res => res?.graphqlApi,
+  );
+
+  getGraphqlApis = crudBuilderFormat<AppSync, 'listGraphqlApis', GraphqlApiAWS[]>(
+    'listGraphqlApis',
+    () => ({}),
+    res => res?.graphqlApis ?? [],
+  );
+
+  updateGraphqlApi = crudBuilder2<AppSync, 'updateGraphqlApi'>('updateGraphqlApi', input => input);
+
+  deleteGraphqlApi = crudBuilder2<AppSync, 'deleteGraphqlApi'>('deleteGraphqlApi', input => input);
+
+  cloud = new Crud2({
+    updateOrReplace: (_a: GraphqlApi, _b: GraphqlApi) => {
+      return 'update';
+    },
+    create: async (apis: GraphqlApi[], ctx: Context) => {
+      const client = (await ctx.getAwsClient()) as AWS;
+      const out = [];
+      for (const api of apis) {
+        const input: CreateGraphqlApiCommandInput = {
+          authenticationType: api.authenticationType,
+          lambdaAuthorizerConfig: api.lambdaAuthorizerConfig,
+          name: api.name,
+          openIDConnectConfig: api.openIDConnectConfig,
+          userPoolConfig: api.userPoolConfig,
+        };
+        const res: GraphqlApiAWS | undefined = await this.createGraphqlApi(client.appSyncClient, input);
+        if (res) {
+          const newApi = this.graphqlApiMapper(res);
+          if (!newApi) continue;
+          await this.module.graphqlApi.db.update(newApi, ctx);
+          out.push(newApi);
+        }
+      }
+      return out;
+    },
+    read: async (ctx: Context, apiId?: string) => {
+      const client = (await ctx.getAwsClient()) as AWS;
+      if (apiId) {
+        const rawApi = await this.getGraphqlApi(client.appSyncClient, apiId);
+        if (!rawApi) return undefined;
+        return this.graphqlApiMapper(rawApi);
+      } else {
+        const rawApis = (await this.getGraphqlApis(client.appSyncClient)) ?? [];
+        const out = [];
+        for (const i of rawApis) {
+          const outApi = this.graphqlApiMapper(i);
+          if (outApi) out.push(outApi);
+        }
+        return out;
+      }
+    },
+    update: async (apis: GraphqlApi[], ctx: Context) => {
+      // if user has modified specific values, restore it. If not, go with update path
+      const client = (await ctx.getAwsClient()) as AWS;
+      const out = [];
+      for (const api of apis) {
+        const cloudRecord = ctx?.memo?.cloud?.GraphqlApi?.[api.name ?? ''];
+        const isUpdate = Object.is(this.module.graphqlApi.cloud.updateOrReplace(cloudRecord, api), 'update');
+        if (isUpdate) {
+          // in case of key fields being modified, restore them
+          if (api.apiId !== cloudRecord.apiId || api.arn !== cloudRecord.arn) {
+            api.apiId = cloudRecord.apiId;
+            api.arn = cloudRecord.arn;
+            await this.module.graphqlApi.db.update(api, ctx);
+            out.push(api);
+          } else {
```

Comment:
hmm, I think this might need to be refactored a little bit. If the user updates the `api_id` and an updateable field like `authenticationType` it will only restore the record but ignore the cloud update. I think what we do in other cases is keep track if an update occurred, if not, check if we need to restore.